docker/dockerfile.merlin

```diff
@@ -144,7 +164,17 @@ RUN rm -rf /repos
 HEALTHCHECK NONE
 CMD ["/bin/bash"]
 
-FROM ${BASE_IMAGE} as base
+FROM ${BASE_IMAGE} as base-amd64
+ONBUILD COPY --chown=1000:1000 --from=triton /usr/lib/x86_64-linux-gnu/libdcgm.so.2 /usr/lib/x86_64-linux-gnu/libdcgm.so.2
+ONBUILD COPY --chown=1000:1000 --from=triton /usr/local/cuda-11.8/targets/x86_64-linux/lib/libcupti.so.11.8 /usr/local/cuda-11.8/targets/x86_64-linux/lib/libcupti.so.11.8
+ONBUILD COPY --chown=1000:1000 --from=triton /opt/tritonserver/backends/fil /opt/tritonserver/backends/fil/
+
+FROM ${BASE_IMAGE} as base-arm64
+ONBUILD COPY --chown=1000:1000 --from=triton /usr/lib/aarch64-linux-gnu/libdcgm.so.2 /usr/lib/aarch64-linux-gnu/libdcgm.so.2
+ONBUILD COPY --chown=1000:1000 --from=triton /usr/local/cuda-11.8/targets/sbsa-linux/lib/libcupti.so.11.8 /usr/local/cuda-11.8/targets/sbsa-linux/lib/libcupti.so.11.8
+RUN echo "Skipping copy of /opt/tritonserver/backends/fil. (Why does this fail on arm64?)"
```

Comment:
Cant these two sets of instructions be condensed? They are running from the same base image. Or am I misunderstanding what is happening here?

Reply:
Same as above, they must be separated, not combined. Lines 167-170 are for amd64 (i.e. x86_64) while lines 172-175 are for arm64 (i.e aarch64). Note how the amd64 filepaths contain `x86_64` while the arm64 filepaths contain `sbsa`.  This is one way to do a conditional copy in Docker where different file paths are copied depending on the target architecture.